packages/dev/core/src/Materials/Node/Blocks/Dual/clipPlanesBlock.ts

```diff
@@ -48,6 +49,27 @@ export class ClipPlanesBlock extends NodeMaterialBlock {
         state._excludeVariableName("fClipDistance5");
         state._excludeVariableName("vClipPlane6");
         state._excludeVariableName("fClipDistance6");
+
+        this._initShaderSourceAsync(state.shaderLanguage);
+    }
+
+    private async _initShaderSourceAsync(shaderLanguage: ShaderLanguage) {
+        this._codeIsReady = false;
+
+        if (shaderLanguage === ShaderLanguage.WGSL) {
+            await import("../../../../ShadersWGSL/ShadersInclude/clipPlaneFragment");
+            await import("../../../../ShadersWGSL/ShadersInclude/clipPlaneFragmentDeclaration");
+            await import("../../../../ShadersWGSL/ShadersInclude/clipPlaneVertex");
+            await import("../../../../ShadersWGSL/ShadersInclude/clipPlaneVertexDeclaration");
+        } else {
+            await import("../../../../Shaders/ShadersInclude/clipPlaneFragment");
+            await import("../../../../Shaders/ShadersInclude/clipPlaneFragmentDeclaration");
+            await import("../../../../Shaders/ShadersInclude/clipPlaneVertex");
+            await import("../../../../Shaders/ShadersInclude/clipPlaneVertexDeclaration");

```

Comment:
general comment for all shader loading - would be better to await a single time to all of them in parallel, in case someone is actually using is in es6 and not UMD. something like:

```javascript
await Promise.all([
import...
import...
import...
]);
```

Reply:
I will change that